plugins/outputs/splunk/README.md

```diff
@@ -0,0 +1,76 @@
+# Splunk Output Plugin
+
+This plugin writes metrics to a [Splunk Metric Index](https://docs.splunk.com/Documentation/Splunk/latest/Indexer/Setupmultipleindexes#Create_metrics_indexes) via a [Splunk HTTP Event Collector (HEC)](http://docs.splunk.com/Documentation/Splunk/latest/Data/UsetheHTTPEventCollector).  This requires an `Authorization Token` which will be created when you setup your Splunk HEC Token.
+
+Notes:
+1. If any point value cannot be converted to a float64, that metric will be skipped.
+2. Metrics are grouped by converting any `_` characters to `.` in the Metric Name.
+
+
+
+### Configuration:
+
+```toml
+# Configuration for Splunk server to send metrics to
+[[outputs.splunk]]
+  ## REQUIRED
+  ## URL of the Splunk Enterprise HEC endpoint (i.e.: http://localhost:8088/services/collector)
+  SplunkUrl = "http://localhost:8088/services/collector"
+
+  ## REQUIRED
+  ## Splunk Authorization Token for sending data to a Splunk HTTPEventCollector (HEC).
+  ##   Note:  This Token should map to a 'metrics' index in Splunk.  
+  AuthString = "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
+
+  ## OPTIONAL:  prefix for metrics keys
+  #prefix = "my.specific.prefix."
+
+  ## OPTIONAL:  whether to use "value" for name of simple fields
+  #simple_fields = false
+
+  ## OPTIONAL:  character to use between metric and field name.  defaults to . (dot)
+  #metric_separator = "."
+
+  ## OPTIONAL:  Convert metric name paths to use metricSeperator character
+  ## When true (default) will convert all _ (underscore) chartacters in final metric name
+  #convert_paths = true
```

Comment:
This doesn't seem like a good idea, at least not as a default. In splunk, the . is supposed to be for namespacing. Lots of telegraf metrics use underscore in field names, and not as namespace boundaries.

Reply:
Updated the default to false, but the option is there if a user wants to convert "_" to "." (or any other metric separator).